chore : Label 컴포넌트 폴더화 및 스토리북 추가

KimDongGyun23 opened the pull request:
## 변경 사항

- 폴더구조 규칙에 의해 Label 컴포넌트를 폴더화
- 스토리북 테스트 추가
- props 타입 변경

<br/>

## 리뷰 필요

`interface`가 아닌 `type`을 사용하는 것을 권유드립니다.

<br/>

첫 번째 이유는 type을 사용하면 함수의 타입을 더 명확하게 정의할 수 있습니다.
`type StringToNumber = (str: string) => number;`

<br/>

두 번째 이유는 유틸리티 타입들이 종종 사용될텐데 이때는 `type`을 사용해야만 합니다.
interface와 type의 혼용보다는 type만으로 일관성을 유지하기 위해서입니다.
```
type User = {
  id: string;
  name: string;
  age: number;
};

type PartialUser = Partial<User>;
```

<br/>

어떠한 방식이라도 하나의 컨벤션을 정해서 사용한다면 더 좋은 프로젝트가 될 것 같습니다. 
자유롭게 의견 남겨주세요 :)

<br/>

close #36

KimDongGyun23 committed:
chore : Label 컴포넌트 폴더화 및 props 타입 변경

폴더구조 규칙에 의해 폴더화 진행
interface 대신 type 사용
props 타입 이름 통일성을 위해 Props로 수정

#36

KimDongGyun23 committed:
feat : Label 컴포넌트 스토리북 추가

#36

kimsuyeon0916 commented:
type 컨벤션을 사용하는 거 저는 좋습니다! 코드의 일관성이 유지될 것 같네용

kimsuyeon0916 commented on src/components/label/Label.tsx:
```diff
@@ -1,11 +1,11 @@
 
-interface LabelProps {
+type Props = {
   children: React.ReactNode
   src?: string
 }
 
-const Label = ({ children, src }: LabelProps) => {
+const Label = ({ children, src }: Props) => {
   return (
     <div className="flex-align gap-2">
       {src && <Image src={src} width={28} height={28} alt="label-icon" />}
```
```typescript
import { type PropsWithChildren } from 'react'
import Image from 'next/image'

type LabelProps = {
  src?: string
}

const Label = ({ children, src }: PropsWithChildren<LabelProps>) => {
  return (
    <div className="flex-align gap-2">
      {src && <Image src={src} width={28} height={28} alt="label-icon" />}
      <p className="subtitle-B">{children}</p>
    </div>
  )
}

export default Label
```

로 작성하시는 거 어떠신가용?!!

KimDongGyun23 replied:
네 해당 부분 수정해놓았습니다. 감사합니다 :)

KimDongGyun23 committed:
chore : Label 타입명 및 속성 변경

Props에서 LabelProps로 이름 변경
children을 ReactNode로 받는 것이 아닌 PropsWithChildren 사용

#36
